packages/stellar_client/lib/src/client.dart

```diff
@@ -0,0 +1,440 @@
+        print("Failed to add trustline for $asset_code.");
+        return false;
+      }
+    } catch (error) {
+      throw Exception("Error funding trustline : $error");
+    }
+  }
+
+  Future<TransactionData?> fetchFundDetails({required String assetCode}) async {
+    try {
+      final response = await http.get(
+        Uri.parse(
+            '${_serviceUrls[_network.toString()]}/transactionfunding_service/conditions'),
+        headers: {'Content-Type': 'application/json'},
+      );
+
+      final body = jsonDecode(response.body);
+      Map<String, dynamic>? details;
+      for (var item in body) {
+        if (item['asset'].startsWith(assetCode)) {
+          details = item;
+          break;
+        }
+      }
+
+      if (details != null) {
+        TransactionData transactionData = TransactionData.fromJson(details);
+        print(transactionData);
+        return transactionData;
+      } else {
+        print('Failed to details for asset: ');
+      }
+
+      return null;
+    } catch (error) {
+      throw Exception("Failed to get fund details, $error");
+    }
+  }
+
+  Future<Operation?> makeFundPaumentOperation(
+      {required String assetCode, required String issuer}) async {
+    TransactionData? transactionData =
+        await fetchFundDetails(assetCode: assetCode);
+    Asset asset = AssetTypeCreditAlphaNum4(assetCode, issuer);
+
+    return PaymentOperationBuilder(
+            transactionData!.feeAccountId, asset, transactionData.feeFixed)
+        .build();
+  }
+
+  Future<Transaction?> buildFundedPaymentTransaction(
+      {required KeyPair sourceKeyPair,
+      required String destinationAddress,
+      required String amount,
+      required String currency,
+      String? memoText}) async {
+    // check that receiver account exists
+    final receiver = await _sdk.accounts.account(accountId);
+
+    // check that asset exists
+    var specificBalance = receiver.balances.firstWhere(
+      (balance) => balance.assetCode == currency,
+      orElse: () {
+        throw Exception('Balance with asset code ${currency} not found.');
+      },
+    );
+
+    final asset = _currencies.currencies[currency];
+    if (asset == null) {
+      throw Exception('${currency} not supported');
+    }
+
+    AccountResponse sourceAccount = await _sdk.accounts.account(accountId);
+
+    Operation? paymentOperation = await makeFundPaumentOperation(
+        assetCode: asset.assetCode, issuer: asset.issuer);
+
+    Asset tftAsset = AssetTypeCreditAlphaNum4(asset.assetCode, asset.issuer);
+
+    Transaction transaction = TransactionBuilder(sourceAccount)
+        .addOperation(paymentOperation!)
+        .addOperation(
+            PaymentOperationBuilder(destinationAddress, tftAsset, amount)
+                .build())
+        .addMemo(memoText != null ? Memo.text(memoText) : Memo.none())
+        .build();
+    return transaction;
+  }
+
+  Future<void> submitFundedTransaction(Transaction fundedTransaction) async {
```

Comment:
is it sending a funded transaction or funding the transaction and sending it?

Reply:
Funds and signs a TFT transaction.